### PR TITLE
fix(*): Compatibility Updates

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: SURF-1519 (epic)

### LGTM's
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* failed.

Built with commit c32525ada3b2c9301400308932c61d33e70cc064

https://app.netlify.com/sites/helix-ui/deploys/5c1283043212c5000b69dc6a

File: src/helix-ui/elements/HXAccordionElement.js [hunk 1]
```diff
@@ -27,12 +27,17 @@ export class HXAccordionElement extends HXElement {
...
         this.panels.forEach(panel => {
             panel.addEventListener('open', this._onPanelOpen);
         });
+
+        // FIXME: initialize on connect
     }
 
     $onDisconnect () {
+        // FIXME: panels may not be present on disconnect
+        // (react disconnects children before the parent)
```
Comment (CITguy) [inline src/helix-ui/elements/HXAccordionElement.js]: NOTE: Need to verify React behavior

Comment (CITguy) [reply src/helix-ui/elements/HXAccordionElement.js]: * Parent React components will have their `componentWillUnmount()` called before any child `disconnectedCallback()` and `componentWillUnmount()` is called. 
    * Disconnect/Mount from the outside-in.
* Children are always present on disconnect.

File: src/helix-ui/elements/HXAccordionElement.js [hunk 2]
```diff
@@ -27,12 +27,17 @@ export class HXAccordionElement extends HXElement {
 
     $onConnect () {
         this.$upgradeProperty('currentPanel');
+        // FIXME: panels are not connected at this point
```
Comment (CITguy) [inline src/helix-ui/elements/HXAccordionElement.js]: Untrue. Children _should_ be present, but Edge/IE require the `$onConnect` to be deferred to the next event loop using `setTimeout(this.$onConnect, 0)` in `HXElement.js`.

File: src/helix-ui/elements/HXDisclosureElement.js [hunk 3]
```diff
@@ -82,6 +84,7 @@ export class HXDisclosureElement extends HXElement {
     get target () {
         if (!this._target) {
             let targetId = this.getAttribute('aria-controls');
+            // FIXME: getRootNode() will not return document context before connect
             this._target = this.getRootNode().getElementById(targetId);
```
Comment (CITguy) [inline src/helix-ui/elements/HXDisclosureElement.js]: Change `getElementById` to `querySelector`

**Comment by CITguy** [comment]
Dev LGTM

File: src/helix-ui/elements/HXTabsetElement.js [hunk 4]
```diff
@@ -25,28 +26,30 @@ export class HXTabsetElement extends HXElement {
     }
 
     $onCreate () {
+        this.$onConnect = defer(this.$onConnect);
```
[the remark this answers is not in the document]
Comment (CITguy) [reply src/helix-ui/elements/HXTabsetElement.js]: Corrects patch from #387 to properly fix #334

**Pull request merged**